ex1/buttonActions.js

```diff
@@ -1,10 +1,13 @@
-function changeColor() {
 
-  if(document.getElementById("demo").style.backgroundColor == "blue")
-   {
-    document.getElementById("demo").style.backgroundColor = "red";
-  }
-  else {
-    document.getElementById("demo").style.backgroundColor = "blue";
-  }
+var i = 1;
+function changeColor() {
+  var colorArray = ['#FF6633', '#CC9999','#4D80CC'];
+    //store
+    sessionStorage.setItem("color", colorArray[i]);
+    // Retrieve
+    document.getElementById("demo").style.backgroundColor = sessionStorage.getItem("color");
+    i++;
```

Comment:
please remove this line and change the next line to:
... ++i  == colorArray.length ....

please read what ++ before a variable does in google